ui/app/js/controllers/user_edit_controller.js

```diff
@@ -1,57 +1,66 @@
 /*global sweetAlert swal */
 angular.module('app').controller('UserPermissionsCtrl',
-function ($scope, $modalInstance, CSRF, Permissions, user, users) {
+function ($scope, $modalInstance, CSRF, Permissions, users, is_edit, user) {
 
   $scope.loading = true;
   $scope.users = users;
   $scope.currentItemTab = 1;
-  $scope.is_edit = true;
-  $scope.original_user = user;
-  $scope.user = angular.copy(user);
+  $scope.is_edit = is_edit;
   $scope.permission = {
     permission: '',
     options_as_json: ''
   };
   $scope.errors = {
     permissions: {}
   };
-  $scope.user.permissions = [];
-  Permissions.getUserPermissions(user.username)
-  .then(function(permissions) {
-    _.forEach(permissions, function(p) {
-      if (p.options) {
-        p.options_as_json = JSON.stringify(p['options']);
-      }
+  if($scope.is_edit){
+    $scope.original_user = user;
+    $scope.user = angular.copy(user);
+    $scope.loading = false;
```

Comment:
You should wait until after all of the network requests have finished to set this back to false (ie: in a .finally() block) - they are the most likely thing to cause loading to fail.